src/Core/Commander/Interfaces/ApiInterface/ApiGlobe.js

```diff
@@ -613,17 +630,38 @@ ApiGlobe.prototype.computeDistance = function computeDistance(p1, p2) {
     return this.scene.getEllipsoid().computeDistance(new GeoCoordinate().copy(p1), new GeoCoordinate().copy(p2));
 };
 
+ApiGlobe.prototype.setSceneLoaded = function computeDistance(callback) {
+    loaded = false;
+    this.callback = () => {
+        callback();
+        this.removeEventListener('globe-loaded', this.callback);
+    };
+    this.addEventListener('globe-loaded', this.callback);
+};
+
 /**
  * Changes the center of the scene on screen to the specified coordinates.
  * <iframe width="100%" height="400" src="//jsfiddle.net/iTownsIGN/x06yhbq6/embedded/" allowfullscreen="allowfullscreen" frameborder="0"></iframe>
  * @constructor
  * @param {coordinates} coordinates - Properties : longitude and latitude
+ * @param      {boolean}  isAnimated  Indicates if animated
+ * @return     {Promise}
  */
-ApiGlobe.prototype.setCenter = function setCenter(coordinates) {
+ApiGlobe.prototype.setCenter = function setCenter(coordinates, isAnimated) {
+    isAnimated = isAnimated || this.isAnimationEnabled();
     eventCenter.oldCenter = this.getCenter();
-    var position3D = this.scene.getEllipsoid().cartographicToCartesian(new GeoCoordinate(coordinates.longitude, coordinates.latitude, 0, UNIT.DEGREE));
-    this.scene.currentControls().setCenter(position3D);
-    this.viewerDiv.dispatchEvent(eventCenter);
+    const position3D = this.scene.getEllipsoid().cartographicToCartesian(new GeoCoordinate(coordinates.longitude, coordinates.latitude, 0, UNIT.DEGREE));
+    position3D.range = coordinates.range;
+    return new Promise((r) => {
```

Comment:
With the above change,  this will become
```js
    return this.scene.currentControls().setCenter(position3D, isAnimated).then(() => {
        this.scene.notifyChange(1);
        return this.globeLoadedDeferred.promise.then(() => {
            this.scene.currentControls().updateCameraTransformation();
            this.viewerDiv.dispatchEvent(eventCenter);
        });
    });
```
And calls to `setCenter().then` would also get executed after this then.